modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/views/table/dnd_table/Cell.tsx

```diff
@@ -50,9 +50,30 @@ const Cell = ({
 	const clientXRef = useRef({current: null});
 
 	useEffect(() => {
+		const CELL_WIDTH = {
+			LARGE: 82,
+			MEDIUM: 75,
+			SMALL: 70,
+		};
+
 		if (columnName && heading && !isFixed && cellRef.current) {
 			const boundingClientRect = cellRef.current.getBoundingClientRect();
 
+			if (boundingClientRect.width < CELL_WIDTH.SMALL) {
+				const fontSize = getComputedStyle(cellRef.current).fontSize;
+
+				boundingClientRect.width += parseFloat(fontSize) * 4;
+			}
+
+			if (
+				boundingClientRect.width > CELL_WIDTH.MEDIUM &&
+				boundingClientRect.width < CELL_WIDTH.LARGE
+			) {
+				const fontSize = getComputedStyle(cellRef.current).fontSize;
+
+				boundingClientRect.width += parseFloat(fontSize) * 2;
+			}
```

Comment:
I chose not to make changes between small and medium since the item-action cell falls in this and would increase its size. This can be seen in dataset menu less than 82 : https://github.com/liferay-frontend/liferay-portal/pull/4313#issuecomment-2234237539

I can change this if we want to only check CELL_WIDTH.SMALL.

Reply:
Action cell is a special case, I think you can exclude it from resizing explicitly in code.

Reply:
I'll test this out and see how other cells behave. Thank you!